src/Http/OpenApi/src/OpenApiGenerator.cs

```diff
@@ -0,0 +1,472 @@
+        var requestBodyContent = new Dictionary<string, OpenApiMediaType>();
+        var isRequired = false;
+
+        if (acceptsMetadata is not null)
+        {
+            foreach (var contentType in acceptsMetadata.ContentTypes)
+            {
+                requestBodyContent[contentType] = new OpenApiMediaType
+                {
+                    Schema = new OpenApiSchema
+                    {
+                        Type = SchemaGenerator.GetOpenApiSchemaType(acceptsMetadata.RequestType ?? requestBodyParameter?.ParameterType)
+                    }
+                };
+            }
+            isRequired = !acceptsMetadata.IsOptional;
+        }
+
+        if (!hasFormOrBodyParameter)
+        {
+            return new OpenApiRequestBody()
+            {
+                Required = isRequired,
+                Content = requestBodyContent
+            };
+        }
+
+        if (requestBodyParameter is not null)
+        {
+            if (requestBodyContent.Count == 0)
+            {
+                var isFormType = requestBodyParameter.ParameterType == typeof(IFormFile) || requestBodyParameter.ParameterType == typeof(IFormFileCollection);
+                var hasFormAttribute = requestBodyParameter.GetCustomAttributes().OfType<IFromFormMetadata>().FirstOrDefault() != null;
+                if (isFormType || hasFormAttribute)
+                {
+                    requestBodyContent["multipart/form-data"] = new OpenApiMediaType
+                    {
+                        Schema = new OpenApiSchema
+                        {
+                            Type = SchemaGenerator.GetOpenApiSchemaType(requestBodyParameter.ParameterType)
+                        }
+                    };
+                }
+                else
+                {
+                    requestBodyContent["application/json"] = new OpenApiMediaType
+                    {
+                        Schema = new OpenApiSchema
+                        {
+                            Type = SchemaGenerator.GetOpenApiSchemaType(requestBodyParameter.ParameterType)
+                        }
+                    }; ;
+                }
+            }
+
+            var nullabilityContext = new NullabilityInfoContext();
+            var nullability = nullabilityContext.Create(requestBodyParameter);
+            var allowEmpty = requestBodyParameter.GetCustomAttributes().OfType<IFromBodyMetadata>().SingleOrDefault()?.AllowEmpty ?? false;
+            var isOptional = requestBodyParameter.HasDefaultValue
+                || nullability.ReadState != NullabilityState.NotNull
+                || allowEmpty;
+
+            return new OpenApiRequestBody
+            {
+                Required = !isOptional,
+                Content = requestBodyContent
+            };
+        }
+
+        return null;
+    }
+
+    private IList<OpenApiTag> GetOperationTags(MethodInfo methodInfo, EndpointMetadataCollection metadata)
```

Comment:
nit: Internal/private APIs should use `List` over `IList`

Reply:
TIL. Fixed.